### PR TITLE
Ea 2428 multiple users entityset

### PR DESCRIPTION
**Pull request by SrishtiSingh-eu**

### CODE REVIEW

File: set-integration-testing/src/test/java/eu/europeana/api/set/integration/web/WebUserSetRestTest.java [hunk 1]
```diff
@@ -114,11 +114,126 @@ public void create_UserSet_400_unauthorized_InvalidJWTToken() throws Exception {
...
+		String result = mockMvc.perform(post(BASE_URL).queryParam(CommonApiConstants.QUERY_PARAM_PROFILE, LdProfiles.MINIMAL.name())
+				.content(requestJson).header(HttpHeaders.AUTHORIZATION, editorUserToken)
+				.header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE))
+				.andExpect(status().is(HttpStatus.CREATED.value())).andReturn().getResponse().getContentAsString();
+
+		String identifier = getSetIdentifier(result);
+		assertNotNull(getSetIdentifier(result));
+		String creator = getSetCreator(result);
+		assertNotNull(creator);
+        assertTrue(StringUtils.contains(creator, getConfiguration().getEntityUserSetUserId()));
```
Comment (gsergiu) [inline set-integration-testing/src/test/java/eu/europeana/api/set/integration/web/WebUserSetRestTest.java:152]: should also check the presence of the user in the contributor list

File: set-integration-testing/src/test/java/eu/europeana/api/set/integration/web/WebUserSetRestTest.java [hunk 2]
```diff
@@ -114,11 +114,126 @@ public void create_UserSet_400_unauthorized_InvalidJWTToken() throws Exception {
...
+
+		WebUserSetImpl userSet = createTestUserSet(ENTITY_USER_SET_REGULAR, editorUserToken);
+		String identifier = userSet.getIdentifier();
+
+		mockMvc.perform(put(BASE_URL+ "{identifier}", identifier)
+				.queryParam(CommonApiConstants.QUERY_PARAM_PROFILE, LdProfiles.STANDARD.name())
+				.content(updateRequestJson).header(HttpHeaders.AUTHORIZATION, editorUserToken)
+				.header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE))
+				.andExpect(status().is(HttpStatus.OK.value()));
+
```
Comment (gsergiu) [inline set-integration-testing/src/test/java/eu/europeana/api/set/integration/web/WebUserSetRestTest.java:180]: should also check the presence of the user in the contributor list

**Review by gsergiu — CHANGES_REQUESTED** [review]
please address the review comments

File: set-web/src/main/java/eu/europeana/set/web/service/impl/BaseUserSetServiceImpl.java [hunk 3]
```diff
@@ -288,17 +295,34 @@ protected boolean hasAdminRights(Authentication authentication) {
...
-		return true;
-	    }
+	public boolean hasEditorRights(Authentication authentication) {
+		if (authentication == null) {
+			return false;
+		}
+		return checkRoles(authentication, Roles.EDITOR.getName());
+	}
+
+	private boolean checkRoles(Authentication authentication, String roleType) {
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/impl/BaseUserSetServiceImpl.java:308]: done

File: set-web/src/main/java/eu/europeana/set/web/service/impl/BaseUserSetServiceImpl.java [hunk 4]
```diff
@@ -288,17 +295,34 @@ protected boolean hasAdminRights(Authentication authentication) {
...
+			// role based authorization
+			String role = iterator.next().getAuthority();
+			if (StringUtils.equalsIgnoreCase(roleType, role)) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	public boolean checkEntityAdminUser(UserSet userSet, Authentication authentication) {
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/impl/BaseUserSetServiceImpl.java:320]: It checks if the user if the Entity set admin/super user. I will rename it to EntityUserSetCreator

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 5]
```diff
@@ -351,32 +350,13 @@ private boolean mustFetchItems(UserSet userSet, LdProfiles profile) {
     }
 
     private void checkVisibilityForUpdate(UserSet existingUserSet, Authentication authentication) throws HttpException {
-	if (hasEditorRole(authentication) && !existingUserSet.isPrivate()) {
+	if (getUserSetService().isEditor(authentication) && !existingUserSet.isPrivate()) {
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:353]: I have removed the method hasEditorRole() from WebUserSetRest and moved it to UserSetService. I will rename it to hasEditorRole()

File: set-web/src/main/java/eu/europeana/set/web/service/impl/BaseUserSetServiceImpl.java [hunk 6]
```diff
@@ -261,10 +263,15 @@ protected boolean isLastPage(int currentPage, int lastPage) {
 
     protected void setDefaults(UserSet newUserSet, Authentication authentication) {
 	Agent user = new WebUser();
-	user.setHttpUrl(getUserId(authentication));
-	user.setNickname(((ApiCredentials) authentication.getCredentials()).getUserName());
-	newUserSet.setCreator(user);
-
+	// if entity set, assign entity admin user as a creator
+    if (StringUtils.equals(newUserSet.getType(), UserSetTypes.ENTITYBESTITEMSSET.getJsonValue())) {
+    	user.setHttpUrl(UserSetUtils.buildCreatorUri(getConfiguration().getEntityUserSetUserId()));
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/impl/BaseUserSetServiceImpl.java:268]: added

File: set-web/src/main/java/eu/europeana/set/web/service/impl/BaseUserSetServiceImpl.java [hunk 7]
```diff
@@ -560,16 +584,43 @@ void validateIsDefinedBy(UserSet webUserSet) throws ParamValidationException, Re
...
 	 * @throws RequestBodyValidationException
 	 */
-	void validateEntityBestItemsSet(UserSet webUserSet) throws ParamValidationException, RequestBodyValidationException {
+	void validateEntityBestItemsSet(UserSet webUserSet, Authentication authentication) throws ParamValidationException,
+			                                                                 RequestBodyValidationException, UserAuthorizationException {
 		if (!isEntityBestItemsSet(webUserSet)) {
 			return;
 		}
+		// only editors, entity admin user and admin user can curate entity user set
+		if (!(hasEditorRights(authentication) || checkEntityAdminUser(webUserSet, authentication) || hasAdminRights(authentication))) {
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/impl/BaseUserSetServiceImpl.java:612]: as discussed have added a checkpermissionForUpdate() method

File: set-web/src/main/java/eu/europeana/set/web/service/impl/UserSetServiceImpl.java [hunk 8]
```diff
@@ -65,7 +66,7 @@
     public UserSet storeUserSet(UserSet newUserSet, Authentication authentication) throws HttpException {
 	setDefaults(newUserSet, authentication);
 
-	validateWebUserSet(newUserSet);
+	validateWebUserSet(newUserSet, authentication);
```
[the remark this answers is not in the document]
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/service/impl/UserSetServiceImpl.java:69]: removed

File: set-integration-testing/src/test/java/eu/europeana/api/set/integration/web/WebUserSetRestTest.java (hunk 1 above)
Comment (SrishtiSingh-eu) [reply set-integration-testing/src/test/java/eu/europeana/api/set/integration/web/WebUserSetRestTest.java:152]: done

File: set-integration-testing/src/test/java/eu/europeana/api/set/integration/web/WebUserSetRestTest.java (hunk 2 above)
Comment (SrishtiSingh-eu) [reply set-integration-testing/src/test/java/eu/europeana/api/set/integration/web/WebUserSetRestTest.java:180]: done

**Pull request merged**